packages/lit-element/src/lit-element.ts

```diff
@@ -97,15 +97,17 @@ let issueWarning: (code: string, warning: string) => void;
 if (DEV_MODE) {
   // Ensure warnings are issued only 1x, even if multiple versions of Lit
   // are loaded.
-  const issuedWarnings: Set<string | undefined> =
-    (globalThis.litIssuedWarnings ??= new Set());
+  globalThis.litIssuedWarnings ??= new Set();
 
   // Issue a warning, if we haven't already.
   issueWarning = (code: string, warning: string) => {
     warning += ` See https://lit.dev/msg/${code} for more information.`;
-    if (!issuedWarnings.has(warning)) {
+    if (
+      !globalThis.litIssuedWarnings!.has(warning) &&
+      !globalThis.litIssuedWarnings!.has(code)
+    ) {
       console.warn(warning);
-      issuedWarnings.add(warning);
+      globalThis.litIssuedWarnings!.add(warning);
```

Comment:
While user now has the ability to disable warnings based on warning code, the issueWarning itself still only disables the warnings it emitted using warning content rather than code.
This is because, in case of reactive-element, the same warning message may be emitted once for each tag name (rather than once for entire app) - that seems like a useful behavior.

https://github.com/lit/lit/blob/6a232e92af9372892c7a916dd3d25947be674ee0/packages/reactive-element/src/reactive-element.ts#L1175-L1178